app-route.html

```diff
@@ -318,8 +317,11 @@
         }
 
         if (this.setProperties) {
-          // atomic update
-          this.setProperties(propertyUpdates, true);
+            //Update properties in a specific order: data, tail and set inactive during update if data changed
+            propertyUpdates.hasOwnProperty("data") && this._setPendingPropertyOrPath("data", propertyUpdates.data, true);
```

Comment:
What about setting without notifying all of the properties:

`this._setPendingPropertyOrPath("data", propertyUpdates.data, false);`

and then call `this.notifyPath('data', propertyUpdates.data)` but in the correct order after all have been set?

Reply:
Not 100% sure this will work; not sure if it will trigger `runEffects`